eth/gasprice/gasprice.go

```diff
@@ -170,6 +188,28 @@ func (oracle *Oracle) SuggestTipCap(ctx context.Context) (*big.Int, error) {
 	if headHash == lastHead {
 		return new(big.Int).Set(lastPrice), nil
 	}
+
+	// If pending txs are less than oracle.congestedThreshold, we consider the network to be non-congested and suggest
+	// a minimal tip cap. This is to prevent users from overpaying for gas when the network is not congested and a few
+	// high-priced txs are causing the suggested tip cap to be high.
+	pendingTxCount, _ := oracle.backend.Stats()
+	if pendingTxCount < oracle.congestedThreshold {
+		// Before Curie (EIP-1559), we need to return the total suggested gas price. After Curie we return 1 wei as the tip cap,
+		// as the base fee is set separately or added manually for legacy transactions.
+		// Set price to one as otherwise tx with a 0 tip might be filtered out by the default mempool config.
+		price := big.NewInt(1)
+		if !oracle.backend.ChainConfig().IsCurie(head.Number) {
+			price = oracle.defaultBasePrice
+		}
+
+		oracle.cacheLock.Lock()
+		oracle.lastHead = headHash
```

Comment:
do we still need to update this, or just early return if pendingTxCount < oracle.congestedThreshold?

Reply:
This is caching the result for the latest block. If we don't do this we need to query the txpool stats for every request to the API, which imo is not necessary. In this way we just do it once per block.